### ISSUE TITLE
Postgres: Unable to use "::" cast syntax together with parameters

### ISSUE THREAD

**Issue by filipsuk** [koxudaxi/local-data-api#81]
**Describe the bug**
I am unable to use PostgreSQL type cast "::" operator together with parameters. This works with AWS Data API.

**To Reproduce**
1. Run a query with a parameter and a type cast. E.g. `SELECT :some_param::VARCHAR`
2. See error `BadRequestException: ERROR: syntax error at or near ":"`

**Expected behavior**
Casts the parameter without an error.

**Additional context**
If you use the other cast syntax, it works:

```
CAST ( expression AS target_type );
```

It also works without a parameter `SELECT 'value'::VARCHAR`.

**Comment by koxudaxi** [comment]
Sorry for the late reply.
local-data-api uses [sqlalchemy](https://www.sqlalchemy.org/) to build a query.
I guess sqlalchemy doesn't support this syntax. 
local-data-api should replace the syntax to `CAST` with `regex`  
I will implement it.

Thank you.

**Comment by koxudaxi** [comment]
What SQL did you run to local-data-api?

I enter simple queries with "::"  to local-data-api.
I used this [docker-compose.yaml](https://github.com/koxudaxi/local-data-api/blob/master/docker-compose-postgres.yml)

It works fine. the value is cast.
```
In [22]: client.execute_statement(resourceArn='arn:aws:rds:us-east-1:12345678901
    ...: 2:cluster:dummy', secretArn='arn:aws:secretsmanager:us-east-1:123456789
    ...: 012:secret:dummy', sql='select 1::VARCHAR', database='test')
Out[22]:
{'ResponseMetadata': {'HTTPStatusCode': 200,
  'HTTPHeaders': {'date': 'Sat, 21 Nov 2020 17:00:56 GMT',
   'server': 'uvicorn',
   'content-length': '62',
   'content-type': 'application/json'},
  'RetryAttempts': 0},
 'numberOfRecordsUpdated': 0,
 'records': [[{'stringValue': '1'}]]}

In [23]: client.execute_statement(resourceArn='arn:aws:rds:us-east-1:12345678901
    ...: 2:cluster:dummy', secretArn='arn:aws:secretsmanager:us-east-1:123456789
    ...: 012:secret:dummy', sql='SELECT datname::VARCHAR FROM pg_database', data
    ...: base='test')
Out[23]:
{'ResponseMetadata': {'HTTPStatusCode': 200,
  'HTTPHeaders': {'date': 'Sat, 21 Nov 2020 17:00:56 GMT',
   'server': 'uvicorn',
   'content-length': '154',
   'content-type': 'application/json'},
  'RetryAttempts': 0},
 'numberOfRecordsUpdated': 0,
 'records': [[{'stringValue': 'postgres'}],
  [{'stringValue': 'test'}],
  [{'stringValue': 'template1'}],
  [{'stringValue': 'template0'}]]}
```

**Comment by filipsuk** [comment]
It looks like you are not casting a parameter value but just a value literal. 

:param::VARCHAR (and pass param=1) - throws error

vs

1::VARCHAR does not throw error

**Comment by koxudaxi** [comment]
@filipsuk 
I'm sorry for the late.
I have released a new version `0.6.0`
The version has been re-written in Kotlin for the JDBC driver.
I believe resolved this problem.

```bash
$ aws rds-data execute-statement --database 'test' \
    --resource-arn $RDS_DATA_API_CLIENT_RESOURCE_ARN \
    --secret-arn $RDS_DATA_API_CLIENT_SECRETARN --endpoint-url http://127.0.0.1:8080 \
    --sql $'select :param::VARCHAR' --parameters '[{"name":"param", "value": {"longValue": 123}}]'
{
    "numberOfRecordsUpdated": 0,
    "records": [
        [
            {
                "stringValue": "123"
            }
        ]
    ]
}
```

**Comment by filipsuk** [comment]
Thank you for the update 🎉 ! I will give it a go later, I will close the issue for now.
